code/modules/religion/rites.dm

```diff
@@ -218,3 +222,191 @@
 		new /obj/item/candle/infinite(altar_turf)
 	playsound(altar_turf, 'sound/magic/fireball.ogg', 50, TRUE)
 	return TRUE
+
+/*********Greedy God**********/
+
+///all greed rites cost money instead
+/datum/religion_rites/greed
+	ritual_length = 5 SECONDS
+	invoke_msg = "Sorry I was late I was just making a shitload of money."
+	var/money_cost = 0
+
+/datum/religion_rites/greed/can_afford(mob/living/user)
+	var/datum/bank_account/account = user.get_bank_account()
+	if(!account)
+		to_chat(user, "<span class='warning'>You need a way to pay for the rite!</span>")
+		return FALSE
+	if(account.account_balance < money_cost)
+		to_chat(user, "<span class='warning'>This rite requires more money!</span>")
+		return FALSE
+	return TRUE
+
+/datum/religion_rites/greed/custom_vending/invoke_effect(mob/living/user, atom/movable/religious_tool)
+	var/datum/bank_account/account = user.get_bank_account()
+	if(!account || account.account_balance < money_cost)
+		to_chat(user, "<span class='warning'>This rite requires more money!</span>")
+		return FALSE
+	account.adjust_money(-money_cost)
+	. = ..()
+
+/datum/religion_rites/greed/vendatray
+	name = "Purchase Vend-a-tray"
+	desc = "Summons a Vend-a-tray. You can use it to sell items!"
+	invoke_msg = "I need a vend-a-tray to make some more money!"
+	money_cost = 300
+
+/datum/religion_rites/greed/vendatray/invoke_effect(mob/living/user, atom/movable/religious_tool)
+	..()
+	var/altar_turf = get_turf(religious_tool)
+	new /obj/structure/displaycase/forsale(altar_turf)
+	playsound(get_turf(religious_tool), 'sound/effects/cashregister.ogg', 60, TRUE)
+	return TRUE
+
+/datum/religion_rites/greed/custom_vending
+	name = "Purchase Personal Vending Machine"
+	desc = "Summons a custom vending machine. You can use it to sell MANY items!"
+	invoke_msg = "If I get a custom vending machine for my products, I'll be RICH!"
+	money_cost = 1000 //quite a step up from vendatray
+
+/datum/religion_rites/greed/custom_vending/invoke_effect(mob/living/user, atom/movable/religious_tool)
+	var/altar_turf = get_turf(religious_tool)
+	new /obj/machinery/vending/custom/greed(altar_turf)
+	playsound(get_turf(religious_tool), 'sound/effects/cashregister.ogg', 60, TRUE)
+	return TRUE
+
```

Comment:
a cool ability for this sect could be something sort of like that cargo drop pod console, but only for personal purchases from your account (and maybe an even greater markup).

do you really need a medkit or a set of tools right about now? well, with the power of money, you can order one from space-amazon on the spot!